### ISSUE TITLE
[Dashboard/K8s] Dashboard fails to start on Ray node running on K8s

### ISSUE THREAD

**Issue by ijrsvt** [ray-project/ray#14220]
### What is the problem?
When running a single ray node on a k8s cluster, the dashboard fails to start.

*Ray version and other system information (Python version, TensorFlow version, OS):*
- Ray 1.1.0

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):
1. Start a K8s pod with `rayproject/ray:1.1.0` as the image and no ports specified.
2. Run `ray start --head --port=6379 --metrics-export-port=8006` in a container on that pod

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

Output:
```
2021-02-18 15:46:44,236 ERROR dashboard.py:210 -- The dashboard on node ians-fun-k8s-pod-0 failed with the following error:
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/dashboard.py", line 197, in <module>
    loop.run_until_complete(dashboard.run())
  File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/dashboard.py", line 100, in run
    await self.dashboard_head.run()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/head.py", line 220, in run
    *(m.run(self.server) for m in modules))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/aiohttp/web.py", line 411, in _run_app
    await site.start()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/aiohttp/web_runner.py", line 128, in start
    reuse_port=self._reuse_port,
  File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 1389, in create_server
    % (sa, err.strerror.lower())) from None
OSError: [Errno 99] error while attempting to bind on address ('::1', 8265, 0, 0): cannot assign requested address
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/dashboard.py", line 197, in <module>
    loop.run_until_complete(dashboard.run())
  File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/dashboard.py", line 100, in run
    await self.dashboard_head.run()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/head.py", line 220, in run
    *(m.run(self.server) for m in modules))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/aiohttp/web.py", line 411, in _run_app
    await site.start()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/aiohttp/web_runner.py", line 128, in start
    reuse_port=self._reuse_port,
  File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 1389, in create_server
    % (sa, err.strerror.lower())) from None
OSError: [Errno 99] error while attempting to bind on address ('::1', 8265, 0, 0): cannot assign requested address
```

**Comment by DmitriGekhtman** [comment]
Could you include the configuration for the pod?

**Comment by DmitriGekhtman** [comment]
this was fixed by adding `--dashboard-host 0.0.0.0` to the args for `ray start`
